pyracf/resource/resource_admin.py

```diff
@@ -589,3 +718,43 @@ def _format_profile(self, result: dict) -> None:
 
         del result["securityResult"]["resource"]["commands"][0]["messages"]
         result["securityResult"]["resource"]["commands"][0]["profiles"] = profiles
+
+    def __validate_access_level(
+        self,
+        success: Union[str, None] = None,
+        failure: Union[str, None] = None,
+        all: Union[str, None] = None,
+    ):
+        access_levels = ["alter", "control", "read", "update"]
+        value_error_text = (
+            "Please specify 'alter', 'control', 'read', or 'update' access "
+            + "for your target attempt."
+        )
+        bad_values = []
+        if success is not None:
+            if success.lower() not in access_levels:
+                bad_values.append(success)
+        if failure is not None:
+            if failure.lower() not in access_levels:
+                bad_values.append(failure)
+        if all is not None:
+            if all.lower() not in access_levels:
+                bad_values.append(all)
+        if len(bad_values) == 0:
```

Comment:
I didn't realize they finally added switch statements to python. I will make this change now.

Reply:
I also refactored code for the "default" case to make it more generic, in case this ever needs to be expanded or generalized to handle more arguments.

Reply:
For the default case, I think you can do this instead to clean things up:

```python
           case _:
                bad_access_levels = [
                    f"'{bad_access_level}'" for bad_access_level in bad_access_levels
                ]
                bad_access_levels[-1] = f"and {bad_access_levels[-1]} "
                value_error_text = (
                    f"{', '.join(bad_access_levels)} are not proper values. {value_error_text}"
                )
```

Reply:
And for `case 2`, you can do:

```python
            case 2:
                value_error_text = (
                    f"'{bad_access_levels[0]}' and '{bad_access_levels[1]}' are not proper "
                    + f"values. {value_error_text}"
                )
```

Reply:
Thank you for the assist in refactoring. I am not always in the habit of thinking in format strings. This is all a lot cleaner. Changes made and the unit tests still pass which is good.

Reply:
My only suggestion now, would just be to further clean up your string building.

For example, I would change `case 1` to:

```python
            case 1:
                value_error_text = (
                    f"'{bad_access_levels[0]}' is not a valid access level. {value_error_text}"
                )
```

I think you can do something similar for your default case as well.

Reply:
default case is currently hamstrunf by the line length requirements at least in the following area (where I believe you are referring)

```python3
    value_error_text = (
                    f"{', '.join(bad_access_levels)}are not valid access levels. "
                    + f"{value_error_text}"
                )
```

works fine, but

```python3
                value_error_text = (
                    f"{', '.join(bad_access_levels)}are not valid access levels. {value_error_text}"
                )
```
leaves a line as 101 characters.

Your instructions for case 1 have been followed though. This was the natural progression of your other comments and it should have been something i did in the last commit.

Reply:
Actually, case1 ended up in a similar situation:

```python3
                value_error_text = f"'{bad_access_levels[0]}' is not a valid access level. {value_error_text}"
```
The above implementation is too long

```python3
                value_error_text = (
                    f"'{bad_access_levels[0]}' is not a valid access level. {value_error_text}"
                )
```
I tried this to get around it but black formatter "optimizes" it into the previous implementation which flake8 then complains about

```python3
                value_error_text = (
                    f"'{bad_access_levels[0]}' is not a valid access level. "
                    + f"{value_error_text}"
                )
```

This then works through both, but in my opinion looks worse than either of the above. If there's an obvious way around this, I am missing it.

Reply:
If this refactoring is putting you over the line limit, then I think you are good.